internal/pkg/term/selector/selector.go

```diff
@@ -149,15 +149,15 @@ func NewConfigSelect(prompt Prompter, store ConfigLister) *ConfigSelect {
 
 // NewWorkspaceSelect returns a new selector that chooses applications and environments from the config store, but
 // services from the local workspace.
-func NewWorkspaceSelect(prompt Prompter, store AppEnvLister, ws WsWorkloadDockerfileLister) *WorkspaceSelect {
+func NewWorkspaceSelect(prompt Prompter, store ConfigLister, ws WsWorkloadDockerfileLister) *WorkspaceSelect {
```

Comment:
What do you think of this instead:
1. Update the `WsWorkloadDockerfileLister` interface to be:
```go
type WorkspaceRetriever interface {
	WsWorkloadLister
	Summary() (*workspace.Summary, error) 
	ListDockerfiles() ([]string, error)
}
```
  This way we can store the application's name (`summary.Application`) into to the `WorkspaceSelect` struct.
2. Use the field in the `Service` and `Job` methods instead of taking an extra parameter.

This has the advantage that a workspace is always associated with an application, so you cannot just pass arbitrary application names. Instead, within the constructor we fetch the app name that's in the workspace and use that for our selection methods. Furthermore, the method signatures won't be updated so we won't have to touch so many different clients.